app/code/Magento/Quote/Model/ValidationRules/ConfigurableProductsValidationRule.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Quote\Model\ValidationRules;
+
+use Magento\Directory\Model\AllowedCountries;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\Quote\Model\Quote;
+
+/**
+ * @inheritdoc
+ */
+class ConfigurableProductsValidationRule implements QuoteValidationRuleInterface
+{
+    /**
+     * @var string
+     */
+    private $generalMessage;
+
+    /**
+     * @var AllowedCountries
+     */
+    private $allowedCountryReader;
+
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @param AllowedCountries $allowedCountryReader
+     * @param ValidationResultFactory $validationResultFactory
+     * @param string $generalMessage
+     */
+    public function __construct(
+        AllowedCountries $allowedCountryReader,
+        ValidationResultFactory $validationResultFactory,
+        string $generalMessage = ''
+    ) {
+        $this->allowedCountryReader = $allowedCountryReader;
+        $this->validationResultFactory = $validationResultFactory;
+        $this->generalMessage = $generalMessage;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(Quote $quote): array
+    {
+        $validationErrors = [];
+        $cartAllowedChildren = [];
+        $cartAllSimpleWithConfigurableParent = [];
+
+        foreach ($quote->getItemsCollection() as $item) {
+            $product = $item->getProduct();
+
+            if (!$item->isDeleted()
+                && $product->getTypeId() === 'configurable'
```

Comment:
Module Quote shouldn't know anything about Configurable products. By adding such references we creating the coupling between modules. Please, move validator to the Configurable product module.

Reply:
Maybe even make sense to create new module QuotoForConfigurableProduct